File: hr_management_app/lib/models/task.dart
class Task {
  final String title;
  final String? description;
  final int assignedById;
  final int assignedToId;
  final String assignedDate;
  final String startDate;
  final String? endDate;
  final String dueDate;
  final String status;
  final String priority;
  final String createdAt;
  final String updatedAt;

  Task({
    required this.title,
    this.description,
    required this.assignedById,
    required this.assignedToId,
    required this.assignedDate,
    required this.startDate,
    this.endDate,
    required this.dueDate,
    required this.status,
    required this.priority,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      title: json['title'] ?? '',
      description: json['description'],
      assignedById: json['assigned_by_id'] ?? 0,
      assignedToId: json['assigned_to_id'] ?? 0,
      assignedDate: json['assigned_date'] ?? '',
      startDate: json['start_date'] ?? '',
      endDate: json['end_date'],
      dueDate: json['due_date'] ?? '',
      status: json['status'] ?? '',
      priority: json['priority'] ?? '',
      createdAt: json['created_at'] ?? '',
      updatedAt: json['updated_at'] ?? '',
    );
  }
}

File: backend/app/Models/Task.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;

class Task extends Model
{
    use HasFactory;

    protected $fillable = [
        'title',
        'description',
        'assigned_by_id',
        'assigned_to_id',
        'assigned_date',
        'start_date',
        'end_date',
        'due_date',
        'status',
        'priority',
    ];

    public function employees(): BelongsToMany
    {
        return $this->belongsToMany(Employee::class, 'employee_task', 'task_id', 'employee_id')
            ->withPivot(['role']);
    }
}